File: BackEnd/src/main/java/org/example/dto/VehiculoRequestDTO.java
package org.example.dto;

import jakarta.validation.constraints.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class VehiculoRequestDTO {
    
    @NotBlank(message = "La matricula es obligatoria")
    @Size(max = 20, message = "La matricula no puede exceder 20 caracteres")
    private String matricula;

    @NotBlank(message = "La marca es obligatoria")
    @Size(max = 50, message = "La marca no puede exceder 50 caracteres")
    private String marca;

    @NotBlank(message = "El modelo es obligatorio")
    @Size(max = 50, message = "El modelo no puede exceder 50 caracteres")
    private String modelo;

    @NotNull(message = "El anio es obligatorio")
    @Min(value = 1900, message = "El anio debe ser mayor a 1900")
    @Max(value = 2027, message = "El anio debe ser menor o igual a 2027")
    private Integer anio;

    private String color;

    @Min(value = 0, message = "El kilometraje no puede ser negativo")
    private Integer kilometraje;

    private String observaciones;
    private String medidasTomadas;
    private String estadoRevision;
    private String imagenBase64;

    @NotNull(message = "El ID del cliente es obligatorio")
    private Long idCliente;

    private Long idMecanico;
}

File: Android/app/src/main/java/com/carlafdez/autolog/data/remote/dto/VehiculoRequestDTO.kt
package com.carlafdez.autolog.data.remote.dto

import com.google.gson.annotations.SerializedName

data class VehiculoRequestDTO(
    @SerializedName("matricula") val matricula: String,
    @SerializedName("marca") val marca: String,
    @SerializedName("modelo") val modelo: String,
    @SerializedName("anio") val anio: Int,
    @SerializedName("color") val color: String?,
    @SerializedName("kilometraje") val kilometraje: Int?,
    @SerializedName("observaciones") val observaciones: String?,
    @SerializedName("medidasTomadas") val medidasTomadas: String?,
    @SerializedName("idCliente") val idCliente: Long?,
    @SerializedName("idMecanico") val idMecanico: Long?
)
